### ISSUE TITLE
Irregular Indexing Time Using Polling Blockhandlers

### ISSUE THREAD

**Issue by itsjerryokolo** [graphprotocol/graph-node#4960]
### Bug report

_Issue from Slack._
Phuture deployed this [repo](https://github.com/olivermehr/subgraph-test
) to Studio. They said the first deployment took 30 mins and now it's taking 5 hours.

cc @lutter 

### IPFS hash

Qmet8fhPnaPGjS6MTJrFkKa8LcQLHXFg8mmnLx7Gc52mn5

### Subgraph name or link to explorer

https://api.studio.thegraph.com/query/39480/subgraph-test/version/latest

### Some information to help us out

- [ ] Tick this box if this bug is caused by a regression found in the latest release.
- [ ] Tick this box if this bug is specific to the hosted service.
- [X] I have searched the issue tracker to make sure this issue is not a duplicate.

### OS information

None

**Comment by olivermehr** [comment]
Block handler is polling every 5k blocks. The indexing is much slower than without the block handler, but more importantly it is slower even before the data source with the block handler kicks in.

**Comment by azf20** [comment]
@olivermehr we had an RPC issue on the studio around the time this came up, have you seen this across multiple deployments with the polling block handler?

**Comment by olivermehr** [comment]
@azf20 yes this is an issue across all deployments i've done in the past few days (including today) where the block handler has been present.

**Comment by incrypto32** [comment]
This appears to be a firehose specific issue. Specifically this, https://github.com/graphprotocol/graph-node/pull/4725#discussion_r1275485953

We are discussing internally how to tackle this.

**Comment by olivermehr** [comment]
Thanks @incrypto32, keep me in the loop with what the go forward plan is.

**Comment by olivermehr** [comment]
@incrypto32 Any update on the above?

**Comment by incrypto32** [comment]
@olivermehr so sorry for not giving an update. This is a firehose limitation, we use firehose on hosted-service and firehose doesn't have the right filter to support this feature. This feature will work perfectly with indexers using RPC instead of firehose. SO you will not see this performance issue when using the decentralized network as many indexers there use the RPC

**Comment by lutter** [comment]
Closing this since we can't fix this; feel free to reopen if more discussion is needed
